homeassistant/components/here_travel_time/sensor.py

```diff
@@ -88,20 +99,67 @@
 NO_ROUTE_ERRORS = ["NGEO_ERROR_GRAPH_DISCONNECTED", "NGEO_ERROR_ROUTE_NO_END_POINT"]
 NO_ROUTE_ERROR_MESSAGE = "HERE could not find a route based on the input"
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+COORDINATE_SCHEMA = vol.Schema(
     {
-        vol.Required(CONF_APP_ID): cv.string,
-        vol.Required(CONF_APP_CODE): cv.string,
-        vol.Required(CONF_DESTINATION): cv.string,
-        vol.Required(CONF_ORIGIN): cv.string,
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-        vol.Optional(CONF_MODE, default=TRAVEL_MODE_CAR): vol.In(TRAVEL_MODE),
-        vol.Optional(CONF_ROUTE_MODE, default=ROUTE_MODE_FASTEST): vol.In(ROUTE_MODE),
-        vol.Optional(CONF_TRAFFIC_MODE, default=False): cv.boolean,
-        vol.Optional(CONF_UNIT_SYSTEM): vol.In(UNITS),
+        vol.Inclusive(CONF_DESTINATION_LATITUDE, "coordinates"): cv.latitude,
+        vol.Inclusive(CONF_DESTINATION_LONGITUDE, "coordinates"): cv.longitude,
     }
 )
 
+PLATFORM_SCHEMA = vol.All(
+    cv.has_at_least_one_key(
+        CONF_DESTINATION_LATITUDE,
+        CONF_DESTINATION_SENSOR,
+        CONF_DESTINATION_DEVICE_TRACKER,
+        CONF_DESTINATION_PERSON,
+        CONF_DESTINATION_ZONE,
+        CONF_DESTINATION_ZONE_FRIENDLY_NAME,
+    ),
+    cv.has_at_least_one_key(
+        CONF_ORIGIN_LATITUDE,
+        CONF_ORIGIN_SENSOR,
+        CONF_ORIGIN_DEVICE_TRACKER,
+        CONF_ORIGIN_PERSON,
+        CONF_ORIGIN_ZONE,
+        CONF_ORIGIN_ZONE_FRIENDLY_NAME,
+    ),
+    PLATFORM_SCHEMA.extend(
+        {
+            vol.Required(CONF_APP_ID): cv.string,
+            vol.Required(CONF_APP_CODE): cv.string,
+            vol.Inclusive(
+                CONF_DESTINATION_LATITUDE, "destination_coordinates"
+            ): cv.latitude,
+            vol.Inclusive(
+                CONF_DESTINATION_LONGITUDE, "destination_coordinates"
+            ): cv.longitude,
+            vol.Exclusive(CONF_DESTINATION_LATITUDE, "destination"): cv.latitude,
```

Comment:
Does it work to set the same key twice, both inclusive and exclusive?

Reply:
For the key `destination` its checks that only one exists (Exlusive). For the key `destination_coordinate` it checks that IF its present, it must be a combination of lat/lon.
It's a bit complicated I know but it was the only way I could think of.

Reply:
Yes, but does it really work?

Reply:
It does. The test all succeed and I tested it in a test instance.